### DISCUSSION TITLE
升级

### DISCUSSION THREAD

**Discussion by yangunang** [apache/doris#13077] [A - General / Q&A]
关于小版本升级这种方案是否可行
1. 添加和当前集群数量一样得新版本be节点到当前集群
2. 通过ALTER SYSTEM DECOMMISSION BACKEND "be_host:be_heartbeat_service_port"; 逐步下线旧版本be节点
3. 当前集群be节点全部升级完完成，然后在按照官方文档升级fe得方式来升级fe节点

**Comment by dataalive** [comment]
虽然可以，但是一般大家也没那么多资源可以准备同等数量的新版本来做数据倒腾，而且decommission的速度不如重启BE替换来的快~

小版本升级可以逐台替换BE后，升级FE中非Master的Follower，观察没问题最后升级Master的Follower即可。
